extensions/kind/src/create-cluster.ts

```diff
@@ -273,6 +383,12 @@ export async function createCluster(
 
     disposeOnDidUpdate.dispose();
 
+    // Wait for CoreDNS to be ready before installing ingress controller
+    // This prevents DNS timeout issues when Contour tries to resolve names
+    if (ingressController) {
+      await waitForCoreDNSReady();
```

Comment:
suggestion(blocking): since this process is a bit complicated, we should add some logging for the user side. Let's pass the `logger` object to `waitForCoreDNSReady`.


```suggestion
      await waitForCoreDNSReady(logger);
```